File: Spring Boot/microservices/distance-service/src/main/java/pl/edu/pg/benchmarking/distance/dto/DistanceRequest.java
package pl.edu.pg.benchmarking.distance.dto;

import lombok.Getter;
import pl.edu.pg.benchmarking.distance.Point;

import java.util.List;
import java.util.function.Function;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class DistanceRequest {

    private static final Pattern COORDINATE_REGEX = Pattern.compile("([0-9]+.[0-9]+), ([0-9]+.[0-9]+)");

    @Getter
    private int placeId1;

    @Getter
    private int placeId2;

    @Getter
    private String coordinates1;

    @Getter
    private String coordinates2;

    private Double parseToLatitude(String coordinate) {
        Matcher matcher = COORDINATE_REGEX.matcher(coordinate);
        matcher.matches();
        return Double.valueOf(matcher.group(1));
    }

    private Double parseToLongitude(String coordinate) {
        Matcher matcher = COORDINATE_REGEX.matcher(coordinate);
        matcher.matches();
        return Double.valueOf(matcher.group(2));
    }

    public static Function<List<DistanceRequest>, List<Point>> dtoToEntityMapper() {
        return request -> request.stream()
                .map(el -> new Point(
                        el.getPlaceId1(), el.getPlaceId2(),
                        el.parseToLatitude(el.getCoordinates1()), el.parseToLongitude(el.getCoordinates1()),
                        el.parseToLatitude(el.getCoordinates2()), el.parseToLongitude(el.getCoordinates2())
                )).toList();
    }

}

File: Flask/microservices/distance-service/website/dto/distance_request.py
import re
from typing import List, Final
from website.point import Point


COORDINATE_REGEX: Final[str] = "([0-9]+.[0-9]+), ([0-9]+.[0-9]+)"


def dto_to_entity_mapper(request: list) -> List[Point]:
    mapped_result = map(lambda places_element: DistanceRequest(places_element).to_point(), request)
    return list(mapped_result)


class DistanceRequest:
    def __init__(self, request_single_place: dict) -> None:
        self.place_id_1 = request_single_place["placeId1"]
        self.place_id_2 = request_single_place["placeId2"]
        self.coordinates_1 = request_single_place["coordinates1"]
        self.coordinates_2 = request_single_place["coordinates2"]

    def parse_to_latitude(self, coordinate: str) -> float:
        result = re.search(COORDINATE_REGEX, coordinate)
        return float(result.group(1))

    def parse_to_longitude(self, coordinate: str) -> float:
        result = re.search(COORDINATE_REGEX, coordinate)
        return float(result.group(2))

    def to_point(self) -> Point:
        return Point(
            self.place_id_1,
            self.place_id_2,
            self.parse_to_latitude(self.coordinates_1),
            self.parse_to_latitude(self.coordinates_2),
            self.parse_to_longitude(self.coordinates_1),
            self.parse_to_longitude(self.coordinates_2)
        )
